modules/jenkins.test.js

```diff
@@ -0,0 +1,111 @@
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+
+const jenkins = require("./jenkins");
+const fs = require("fs");
+
+// #endregion Imports
+
+// -----------------------------------------------------------------------------------------
+// #region Tests
+// -----------------------------------------------------------------------------------------
+
+describe("jenkins", () => {
+    describe("configureCredentials", () => {
+        it("when url is empty then returns false", () => {
+            // Arrange
+            const url = "";
+            const username = "aaa";
+            const token = "aaa";
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+            // Assert
+            expect(result).toBe(false);
+        });
+
+        it("when username is empty then returns false", () => {
+            // Arrange
+            const url = "aaa";
+            const username = "";
+            const token = "aaa";
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+            // Assert
+            expect(result).toBe(false);
+        });
+
+        it("when token is empty then returns with errors", () => {
```

Comment:
How are we defining errors here? If it's meant to refer to the `echo.error` calls, I would add in a spy here and assert that it was called. Otherwise, the test name should be updated similar to the above one.